core/src/main/java/org/mule/runtime/core/internal/routing/result/CompositeRoutingException.java

```diff
@@ -76,8 +76,8 @@ public String getDetailedMessage() {
       builder.append(super.getDetailedMessage());
       // get detailed information about exceptions that make up composite exception
       builder.append(lineSeparator()).append(MESSAGE_SUB_TITLE).append(lineSeparator());
-      for (Entry<String, Pair<Error, EventProcessingException>> entry : detailedFailures.entrySet()) {
-        MuleException muleException = getRootMuleException(entry.getValue().getSecond());
```

Comment:
Why are you removing the getRootMuleException call here?

Reply:
it is what is suggested as the fix in the ticket:
The only usage we seem to be making from them is [here](https://github.com/mulesoft/mule/blob/57ba048eee9d2fb6f2f16fbdcd77be4e866f4c4c/core/src/main/java/org/mule/runtime/core/internal/routing/result/CompositeRoutingException.java#L80), when trying to build the detailed message string. And we are just unwrapping them and taking the root MuleException from them. We might just as well do that when [creating](https://github.com/mulesoft/mule/blob/57ba048eee9d2fb6f2f16fbdcd77be4e866f4c4c/core/src/main/java/org/mule/runtime/core/privileged/routing/RoutingResult.java#L47) the RoutingResult so it can also be converted to JSON.